internal/policy/policy.go

```diff
@@ -855,6 +854,67 @@ func (s *State) getRootVerifier() (*Verifier, error) {
 	}, nil
 }
 
+// Apply takes valid changes from the policy staging ref, and fast-forward
+// merges it into the policy ref. Apply only takes place if the latest state on
+// the policy staging ref is valid. This prevents invalid changes to the policy
+// taking affect, and allowing new changes, that until signed by multiple users
+// would be invalid to be made, by utilizing the policy staging ref.
+func Apply(ctx context.Context, repo *git.Repository, signRSLEntry bool) error {
+	_, err := LoadCurrentState(ctx, repo, PolicyStagingRef)
+	if err != nil {
+		return fmt.Errorf("failed to load current state: %w", err)
+	}
+
+	// Get the reference for the PolicyRef
+	policyRef, err := repo.Reference(plumbing.ReferenceName(PolicyRef), true)
+	if err != nil {
+		return fmt.Errorf("failed to get policy reference %s: %w", PolicyRef, err)
+	}
+
+	// Get the reference for the PolicyStagingRef
+	policyStagingRef, err := repo.Reference(plumbing.ReferenceName(PolicyStagingRef), true)
+	if err != nil {
+		return fmt.Errorf("failed to get policy staging reference %s: %w", PolicyStagingRef, err)
+	}
+
+	// Check if the PolicyStagingRef is ahead of PolicyRef (fast-forward)
```

Comment:
To fail early, we should also be performing this check when loading the staging area as we may be mutating the policy...